client/src/plugins/error-tracking/ErrorTracking.js

```diff
@@ -0,0 +1,187 @@
+
+export default class ErrorTracking extends PureComponent {
+
+  constructor(props) {
+    super(props);
+
+    // OVERRIDE_SENTRY_DSN flag is useful for development.
+    this.SENTRY_DSN = Flags.get(OVERRIDE_SENTRY_DSN) || SENTRY_DSN;
+
+    // Setting this here so that we can mock later if necessary.
+    this.SCHEDULE_TIME = ONE_MINUTE_MS;
+
+    // Setting this here so that we can mock later if necessary.
+    this._sentry = Sentry;
+
+    this._isInitialized = false;
+  }
+
+  async componentDidMount() {
+    const { result, msg } = await this.canInitializeSentry();
+
+    // -> The user may turn on / off error reporting on the run/
+    // -> The user may never actually restart the modeler.
+    // That's why we'll schedule a check and turn on / off Sentry if necessary.
+    this.scheduleCheck();
+
+    if (!result) {
+
+      return log('Cannot initialize: ' + msg);
+    }
+
+    this.initializeSentry();
+  }
+
+  async initializeSentry() {
+
+    const { config, _getGlobal } = this.props;
+
+    const editorID = await config.get(EDITOR_ID_CONFIG_KEY) || NON_EXISTENT_EDITOR_ID;
+
+    const releaseTag = Metadata.data.version;
+
+    try {
+
+      // We need to send a message to backend to initialize Sentry in the main process.
+      // Note that Sentry in renderer process (browser JS) and main process (node)
+      // has to be handled separately -> they have separate source maps and their Sentry
+      // modules are different (sentry-browser and sentry-node).
+      const backend = _getGlobal('backend');
+      backend.send('sentry:initialize', { dsn: this.SENTRY_DSN, releaseTag, editorID });
+
+      // Source map uploaded to Sentry from WebPack is tagged with the
+      // version number in package.json which is supposed to be the same
+      // with Metadata.data.version (except for dev environments
+      // - we don't initialize dev in Sentry.)
+      this._sentry.init({
+        dsn: this.SENTRY_DSN,
+        release: releaseTag
+      });
+
+      // OS information already exists by default in Sentry.
+      // We'll set editor ID and Camunda Modeler version.
+      this._sentry.configureScope(scope => { scope.setTag('editor-id', editorID); });
+
+      const { subscribe } = this.props;
+
+      // Send handled errors to Sentry.
+      subscribe('app.error-handled', (error) => {
+        if (!this._isInitialized) {
+          return;
+        }
+
+        this._sentry.captureException(error);
+      });
+
+      log('Initialized');
+
+      this._isInitialized = true;
+
+    } catch (err) {
+
+      log('Cannot initialize: ', err);
+    }
+  }
+
+  closeSentry() {
+
+    // tell backend to close Sentry-node instance.
+    const { _getGlobal } = this.props;
+    const backend = _getGlobal('backend');
+    backend.send('sentry:close');
+
+    this._sentry.close();
+    this._isInitialized = false;
+
+    log('Closed Sentry.');
+  }
+
+  async canInitializeSentry() {
+
+    if (!this.SENTRY_DSN) {
```

Comment:
We should also test for `Flags.get(DISABLE_REMOTE_INTERACTION)` here.

That one should disable _any_ remote interaction.

Reply:
Done via 2d5b5fb95fdf1be4febfa61bd941985410e25237